File: call-notes-to-pedigree/node/src/extractedFamily.ts
/**
 * Domain value objects returned by the LLM extractor.
 * Mirrors the fields the `family-history-intake-form` demo captures so the
 * writer can reuse the same add-relative orchestration.
 */

export type BiologicalSex = 'female' | 'male' | 'unknown';

export type SiblingRelation = 'sister' | 'brother' | 'half_sister' | 'half_brother';

export interface ProbandEntry {
  readonly displayName: string;
  readonly biologicalSex: BiologicalSex;
  readonly yearOfBirth?: number;
  readonly notes?: string;
}

export interface RelativeEntry {
  readonly displayName: string;
  readonly yearOfBirth?: number;
  readonly notes?: string;
}

export interface SiblingEntry {
  readonly displayName: string;
  readonly relation: SiblingRelation;
  readonly yearOfBirth?: number;
  readonly notes?: string;
}

export interface ExtractedFamily {
  readonly proband: ProbandEntry;
  readonly mother?: RelativeEntry;
  readonly father?: RelativeEntry;
  readonly maternalGrandmother?: RelativeEntry;
  readonly maternalGrandfather?: RelativeEntry;
  readonly paternalGrandmother?: RelativeEntry;
  readonly paternalGrandfather?: RelativeEntry;
  readonly siblings: readonly SiblingEntry[];
}

export function siblingBiologicalSex(relation: SiblingRelation): 'female' | 'male' {
  return relation === 'sister' || relation === 'half_sister' ? 'female' : 'male';
}

File: call-notes-to-pedigree/python/src/call_notes_to_pedigree/extracted_family.py
"""Domain value objects returned by the LLM extractor.

These mirror the fields the ``family-history-intake-form`` demo captures
by hand, so the writer can reuse the same add-relative orchestration.
"""

from __future__ import annotations

from dataclasses import dataclass, field
from enum import Enum


class BiologicalSex(str, Enum):
    FEMALE = "female"
    MALE = "male"
    UNKNOWN = "unknown"


class SiblingRelation(str, Enum):
    SISTER = "sister"
    BROTHER = "brother"
    HALF_SISTER = "half_sister"
    HALF_BROTHER = "half_brother"


@dataclass(frozen=True)
class ProbandEntry:
    display_name: str
    biological_sex: BiologicalSex
    year_of_birth: int | None = None
    notes: str | None = None


@dataclass(frozen=True)
class RelativeEntry:
    display_name: str
    year_of_birth: int | None = None
    notes: str | None = None


@dataclass(frozen=True)
class SiblingEntry:
    display_name: str
    relation: SiblingRelation
    year_of_birth: int | None = None
    notes: str | None = None

    @property
    def biological_sex(self) -> BiologicalSex:
        if self.relation in (SiblingRelation.SISTER, SiblingRelation.HALF_SISTER):
            return BiologicalSex.FEMALE
        return BiologicalSex.MALE


@dataclass(frozen=True)
class ExtractedFamily:
    proband: ProbandEntry
    mother: RelativeEntry | None = None
    father: RelativeEntry | None = None
    maternal_grandmother: RelativeEntry | None = None
    maternal_grandfather: RelativeEntry | None = None
    paternal_grandmother: RelativeEntry | None = None
    paternal_grandfather: RelativeEntry | None = None
    siblings: tuple[SiblingEntry, ...] = field(default_factory=tuple)
